File: x-fly/src/main/java/com/xia/fly/utils/FragmentUtils.java
package com.xia.fly.utils;

import android.os.Bundle;
import android.support.v4.app.Fragment;

import me.yokeyword.fragmentation.ISupportFragment;

/**
 * @author xia
 * @date 2018/7/10.
 */
@SuppressWarnings("WeakerAccess")
public final class FragmentUtils {

    public static <T extends ISupportFragment> T newInstance(final Class cls) {
        return newInstance(cls, null);
    }

    @SuppressWarnings("unchecked")
    public static <T extends ISupportFragment> T newInstance(
            final Class cls, final Bundle bundle) {
        try {
            final T t = (T) cls.newInstance();
            if (bundle != null && !bundle.isEmpty()) {
                ((Fragment) t).setArguments(bundle);
                t.putNewBundle(bundle);
            }
            return t;
        } catch (IllegalAccessException ignored) {
        } catch (InstantiationException ignored) {
        }
        return null;
    }
}

File: fly/src/main/java/com/xia/fly/utils/FragmentUtils.kt
package com.xia.fly.utils

import android.os.Bundle

import androidx.fragment.app.Fragment
import me.yokeyword.fragmentation.ISupportFragment

/**
 * @author xia
 * @date 2018/7/10.
 */
@Suppress("UNCHECKED_CAST")
object FragmentUtils {

    @JvmStatic
    fun <T : ISupportFragment> newInstance(cls: Class<out ISupportFragment>): T? {
        return newInstance(cls, null)
    }

    @JvmStatic
    fun <T : ISupportFragment> newInstance(cls: Class<out ISupportFragment>, bundle: Bundle?): T? {
        try {
            val t = cls.newInstance() as? T
            if (t != null) {
                return newInstance(t, bundle)
            }
        } catch (ignored: IllegalAccessException) {
        } catch (ignored: InstantiationException) {
        }
        return null
    }

    @JvmStatic
    fun <T : ISupportFragment> newInstance(fragment: ISupportFragment, bundle: Bundle?): T {
        if (bundle != null && !bundle.isEmpty) {
            (fragment as Fragment).arguments = bundle
            fragment.putNewBundle(bundle)
        }
        return fragment as T
    }
}
